### DISCUSSION TITLE
How do we install libcharon Plugins by pkg

### DISCUSSION THREAD

**Discussion by TomHsiung** [strongswan/strongswan#1960] [General]
The image of FreeBSD provided by the VPS service does not include port system and has no kernel conf files on the disk. I installed strongSwan via pkg command. But I also wish to install a plugin called `bypass-lan`.

Unlike Ubuntu, FreeBSD pkg does not find the package called the `libcharon-extauth-plugins`. Therefore, how do we get the plugin? Thanks.

**Comment by tobiasbrunner** [comment]
I don't think plugins are packaged individually, nor that packages with different options are available (except maybe for different platforms). So to build with different options/plugins, you have to build from the Port (to see the options the package was built with, you can use `pkg options strongswan`). However, note that the Port currently doesn't have an option to build the _bypass-lan_ plugin.

**Reply by TomHsiung** [reply]
Sound like FreeBSD cannot enable the `bypass-lan` plugin.
